aodncore/pipeline/wfs.py

```diff
@@ -0,0 +1,83 @@
+import json
+from collections import OrderedDict
+
+from owslib.etree import etree
+from owslib.fes import PropertyIsEqualTo
+from owslib.wfs import WebFeatureService
+
+from ..util import IndexedSet
+
+__all__ = [
+    'WfsBroker',
+    'get_filter_for_file_url',
+    'ogc_filter_to_string'
+]
+
+
+def ogc_filter_to_string(ogc_filter):
+    """Convert an OGC filter object into it's XML string representation
+
+    :param ogc_filter: OGC filter object
+    :return: XML string
+    """
+    try:
+        return etree.tostring(ogc_filter.toXML()).decode('utf-8')
+    except AttributeError:
+        raise TypeError('parameter must be a subclass of OgcExpression')
+
+
+def get_filter_for_file_url(file_url):
+    """Return OGC filter XML to query for a single file_url
+
+    :param file_url: URL string
+    :return: OGC XML filter string
+    """
+    file_url_filter = PropertyIsEqualTo(propertyname='file_url', literal=file_url)
+    return ogc_filter_to_string(file_url_filter)
+
+
+class WfsBroker(object):
+    """Simple state query interface, to provide user friendly access for querying existing Pipeline state
+    """
+
+    def __init__(self, wfs_url, version='1.0.0'):
+        self._wfs = WebFeatureService(wfs_url, version=version)
+
+    @property
+    def wfs(self):
+        """Read-only property to access the instantiated WebFeatureService object directly
+
+        :return: WebFeatureService instance
+        """
+        return self._wfs
+
+    def _wfs_getfeature_json(self, *args, **kwargs):
```

Comment:
Yes, good idea. I was thinking it would be good to add a few common/useful helper methods to cover the majority of cases, and then assume anything more advanced can just access `self.state_query.wfs` to make arbitrary requests. If the requests turn out to be not so arbitrary, they could turn into helper methods in their own right in core